backend/utils/ua.py

```diff
@@ -0,0 +1,65 @@
+import logging
+import operator
+
+import user_agents
+import yaml
+
+
+def load_browsers_list(fname="browsers.yml"):
+    with open(fname) as f:
+        _data = yaml.load(f)
+    blacklist = _data["blacklist"]
+    return blacklist
+
+
+default_blacklisted_browsers = load_browsers_list()
+
+ops = {
+    "eq": operator.eq,
+    "ge": operator.ge,
+    "gt": operator.gt,
+    "le": operator.le,
+    "lt": operator.lt,
+}
+
+
+def _getattrs(obj, *keys):
+    for key in keys:
+        obj = getattr(obj, key)
+    return obj
+
+
+def _cmp(key):
+    op = "eq"
+    if "__" in key:
+        key, op = key.split("__", 1)
+    return key, ops[op]
+
+
+def _match(ua, browser):
```

Comment:
There was a lot of doctests in the original code. Can you either:

- Include them here
- Rewrite to use pytest?